cpp/src/arrow/flight/sql/server_test.cc

```diff
@@ -0,0 +1,783 @@
+  std::vector<std::string> table_types{"index"};
+
+  ASSERT_OK_AND_ASSIGN(
+      auto flight_info,
+      sql_client->GetTables({}, nullptr, nullptr, nullptr, false, table_types));
+
+  ASSERT_OK_AND_ASSIGN(auto stream,
+                       sql_client->DoGet({}, flight_info->endpoints()[0].ticket));
+
+  std::shared_ptr<Table> table;
+  ASSERT_OK(stream->ReadAll(&table));
+
+  ASSERT_TRUE(table->schema()->Equals(SqlSchema::GetTablesSchema()));
+
+  ASSERT_EQ(table->num_rows(), 0);
+}
+
+TEST(TestFlightSqlServer, TestCommandGetTablesWithUnexistenceTableTypeFilter) {
+  std::vector<std::string> table_types{"table"};
+
+  ASSERT_OK_AND_ASSIGN(
+      auto flight_info,
+      sql_client->GetTables({}, nullptr, nullptr, nullptr, false, table_types));
+
+  ASSERT_OK_AND_ASSIGN(auto stream,
+                       sql_client->DoGet({}, flight_info->endpoints()[0].ticket));
+
+  std::shared_ptr<Table> table;
+  ASSERT_OK(stream->ReadAll(&table));
+
+  ASSERT_OK_AND_ASSIGN(auto catalog_name, MakeArrayOfNull(utf8(), 3))
+  ASSERT_OK_AND_ASSIGN(auto schema_name, MakeArrayOfNull(utf8(), 3))
+  std::shared_ptr<Array> table_name;
+  ArrayFromVector<StringType, std::string>(
+      {
+          "foreignTable",
+          "intTable",
+          "sqlite_sequence",
+      },
+      &table_name);
+  std::shared_ptr<Array> table_type;
+  ArrayFromVector<StringType, std::string>({"table", "table", "table"}, &table_type);
+
+  const std::shared_ptr<Table>& expected_table = Table::Make(
+      SqlSchema::GetTablesSchema(), {catalog_name, schema_name, table_name, table_type});
+
+  AssertTablesEqual(*expected_table, *table);
+}
+
+TEST(TestFlightSqlServer, TestCommandGetTablesWithIncludedSchemas) {
+  std::vector<std::string> table_types;
+
+  std::string table_filter_pattern = "int%";
+  ASSERT_OK_AND_ASSIGN(auto flight_info,
+                       sql_client->GetTables({}, nullptr, nullptr, &table_filter_pattern,
+                                             true, table_types));
+
+  ASSERT_OK_AND_ASSIGN(auto stream,
+                       sql_client->DoGet({}, flight_info->endpoints()[0].ticket));
+
+  std::shared_ptr<Table> table;
+  ASSERT_OK(stream->ReadAll(&table));
+
+  ASSERT_OK_AND_ASSIGN(auto catalog_name, MakeArrayOfNull(utf8(), 1))
+  ASSERT_OK_AND_ASSIGN(auto schema_name, MakeArrayOfNull(utf8(), 1))
+  std::shared_ptr<Array> table_name;
+  std::shared_ptr<Array> table_type;
+  ArrayFromVector<StringType, std::string>({"intTable"}, &table_name);
+  ArrayFromVector<StringType, std::string>({"table"}, &table_type);
+
+  const std::shared_ptr<Schema> schema_table = arrow::schema(
+      {arrow::field("id", int64(), true), arrow::field("keyName", utf8(), true),
+       arrow::field("value", int64(), true), arrow::field("foreignId", int64(), true)});
+
+  const arrow::Result<std::shared_ptr<Buffer>>& value =
+      ipc::SerializeSchema(*schema_table);
+  std::shared_ptr<Buffer> schema_buffer;
+
+  ASSERT_OK_AND_ASSIGN(schema_buffer, value);
```

Comment:
Why don't you directly write `ASSERT_OK_AND_ASSIGN(auto schema_buffer, ipc::SerializeSchema(*schema_table));`?